plugins/linux-drm/fu-linux-drm-plugin.c

```diff
@@ -0,0 +1,138 @@
+/*
+ * Copyright (C) 2023 Richard Hughes <richard@hughsie.com>
+ *
+ * SPDX-License-Identifier: LGPL-2.1+
+ */
+
+#include "config.h"
+
+#include "fu-linux-drm-plugin.h"
+
+struct _FuLinuxDrmPlugin {
+	FuPlugin parent_instance;
+	GPtrArray *devices;
+	guint idle_id;
+};
+
+G_DEFINE_TYPE(FuLinuxDrmPlugin, fu_linux_drm_plugin, FU_TYPE_PLUGIN)
+
+#define FU_LINUX_DRM_PLUGIN_REFRESH_DELAY 2 /* s */
+
+static FuDisplayState
+fu_linux_drm_plugin_get_display_state(FuLinuxDrmPlugin *self)
+{
+	FuDisplayState display_state = FU_DISPLAY_STATE_DISCONNECTED;
+
+	/* no devices detected */
+	if (self->devices->len == 0)
+		return FU_DISPLAY_STATE_UNKNOWN;
+
+	/* any connected display is good enough */
+	for (guint i = 0; i < self->devices->len; i++) {
+		FuDevice *device = g_ptr_array_index(self->devices, i);
+		const gchar *status =
+		    fu_udev_device_get_sysfs_attr(FU_UDEV_DEVICE(device), "status", NULL);
+		if (g_strcmp0(status, "connected") == 0) {
+			display_state = FU_DISPLAY_STATE_CONNECTED;
+			break;
+		}
+	}
+	return display_state;
+}
+
+static gboolean
+fu_linux_drm_plugin_get_display_state_idle_cb(gpointer user_data)
+{
+	FuLinuxDrmPlugin *self = FU_LINUX_DRM_PLUGIN(user_data);
+	FuContext *ctx = fu_plugin_get_context(FU_PLUGIN(self));
+	fu_context_set_display_state(ctx, fu_linux_drm_plugin_get_display_state(self));
+	self->idle_id = 0;
+	return FALSE;
+}
+
+static void
+fu_linux_drm_plugin_get_display_state_in_idle(FuLinuxDrmPlugin *self)
+{
+	if (self->idle_id != 0)
+		g_source_remove(self->idle_id);
+	self->idle_id = g_timeout_add_seconds(FU_LINUX_DRM_PLUGIN_REFRESH_DELAY,
```

Comment:
Is there no uevent to use? 

This makes me wonder if maybe it's better to use a drm ioctl call than parsing the sysfs.

Reply:
There is a uevent: https://dri.freedesktop.org/docs/drm/gpu/drm-uapi.html#sysfs-support.  It's a lot cleaner to register to that IMO.

Reply:
Ohh we use that now -- that's the `plugin_class->backend_device_changed` vfunc -- the only reason I added the idle step is at startup if you have 7 devices we'd re-calculate the global state for each device add -- so you'd have, 1, 1+2, 1+2+3, 1+2+3+4 all the way up to 7. The other way (arguably cleaner) is to have a new plugin vfunc of "i've done all the device probing, now recalculate global state" and remove all the timeout stuff.

Reply:
Ah, I missed that.  Yeah that does sound much cleaner, thanks!

Reply:
https://github.com/fwupd/fwupd/pull/6198

Reply:
Fixed, and rebased.